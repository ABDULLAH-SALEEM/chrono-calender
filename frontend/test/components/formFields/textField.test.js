import React from "react";
import { render, screen } from "@testing-library/react";
import TextField from "../../../src/components/formFields/textField";

describe("TextField", () => {
  it("renders without crashing", () => {
    render(<TextField label="Test Label" value="" onChange={() => {}} />);
    expect(screen.getByRole("textbox")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <TextField label="Test Label" value="" onChange={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
