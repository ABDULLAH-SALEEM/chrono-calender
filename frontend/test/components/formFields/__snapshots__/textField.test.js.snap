// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-cmpglg-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1pzfmz2-MuiInputBase-input-MuiOutlinedInput-input"
          id="«r1»"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1nf2c5d-MuiNotchedOutlined-root"
          >
            <span
              aria-hidden="true"
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
